FirebaseStorage/Sources/StorageTask.swift

```diff
@@ -29,15 +34,66 @@ import FirebaseStorageInternal
    * An immutable view of the task and associated metadata, progress, error, etc.
    */
   @objc public var snapshot: StorageTaskSnapshot {
-    return StorageTaskSnapshot(task: self)
+    objc_sync_enter(StorageTask.self)
```

Comment:
It'd be interesting to see if there's a Swift-y alternative to `objc_sync_` or `@synchronized` but this is fine too.

Reply:
I suspect there is